src/kerberos.cc

```diff
@@ -37,6 +37,13 @@ std::string ToStringWithNonStringAsEmpty(Napi::Value value) {
     return value.As<String>();
 }
 
+int BooleanToIntWithNonIntAsFalse(Napi::Value value) {
+    if (!value.IsBoolean()) {
+        throw Error::New(value.Env(), "Expected a boolean value");
+    }
+    return value.As<Boolean>().Value() ? 1 : 0;
```

Comment:
At the very least, we should be able to unit test the error scenario, right?  Tests for 

1. when no `protect` is provided, no error is thrown
2. when `protect` is provided as a boolean, no error is thrown
3. when `protect` is provided and its not a boolean, we throw an error